src/backend/question/views.py

```diff
@@ -40,8 +49,44 @@ def get_queryset(self):
             queryset = queryset.filter(category=category)
         if keypoint is not None:
             queryset = queryset.filter(keypoint=keypoint)
+        if solution_type is not None and solution_type != 'all':
+            queryset = queryset.filter(type=solution_type)
         return queryset
 
+    def get_queryset(self):
```

Comment:
Instead of rewrite `get_queryset`, we can use [action](https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing) to implement the API just for recommend solution.